### PR TITLE
Remove DB dependency from get_all_node_results

### PR DESCRIPTION
**Pull request by scottwn**
- [ ] I have added the tests to cover my changes.
- [x] I have updated the documentation and CHANGELOG accordingly.
- [x] I have read the CONTRIBUTING document.

### CODE REVIEW

**Comment by FyzHsn** [comment]
@scottwn These tests need to be updated, but otherwise LGTM. I'll approve this PR once the tests get fixed

**Comment by scottwn** [comment]
@FyzHsn I see in this commit you changed the assertion in `test_get_node_error` from `None` to `""`. Do you remember the reason for this? Can I change it back to `None`?
https://github.com/AgnostiqHQ/covalent/commit/67f5febd14bfeb4c9b115628c947c25b55743682#diff-011d90cacd1cb15b39a4a992517b7781ce550ee5a0a0b4401f7d97c58d7b8b0b

**Review by FyzHsn — DISMISSED** [review]

**Review by FyzHsn — APPROVED** [review]

**Pull request merged**
